sycl/include/sycl/queue.hpp

```diff
@@ -141,6 +141,21 @@ class __SYCL_EXPORT SubmissionInfo {
 };
 
 } // namespace v1
+
+template <typename KernelName = detail::auto_name, typename PropertiesT,
+          typename KernelType, int Dims>
+event submit_kernel_direct_with_event(
+    const queue &Queue, PropertiesT Props, nd_range<Dims> Range,
```

Comment:
I wonder if we can/should pass `nd_range<>` by reference here? To avoid unnecessary data copy.